src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -410,10 +412,10 @@ namespace
             {
                 // The appleseed material does not exist yet, let the material plugin create it.
                 material_info.m_name =
-                    make_unique_name(assembly.materials(), wide_to_utf8(mtl->GetName()) + "_mat");
-                assembly.materials().insert(
+                    make_unique_name(parent_assembly->materials(), wide_to_utf8(mtl->GetName()) + "_mat");
+                parent_assembly->materials().insert(
```

Comment:
I guess it's intentional that when a material is created, it's created in the "parent assembly" and not in `assembly` since it allows more sharing, is that correct?

Reply:
If you confirm this then we can merge the PR.

Reply:
Yes, parent_assembly should be root assembly in cases of nested assembly.